### DISCUSSION TITLE
What does the Deterministic flag in the functions editor do?

### DISCUSSION THREAD

**Discussion by ve3meo** [pawelsalawa/sqlitestudio#4839] [Q&A] [answered]
I've turned the flag off and on for a scalar function with no apparent effect. Searching for "deterministic" returned the following among many instances in code:
`ADDED: #4050 Functions Editor allows marking functions as deterministic (thus using them for GENERATED columns).`
The results from the function are displayed in a column regardless of the flag's setting.

**Comment by pawelsalawa** [comment, marked as answer]
It's a feature of SQLite. See https://sqlite.org/c3ref/c_deterministic.html#sqlitedeterministic
Basically, if you want to use a function in `GENERATED` column constraint, such function has to be deterministic and should actually comply with being deterministic (again - see link above for details).
